src/index.js

```diff
@@ -21,30 +23,85 @@ import {
 
 export default class ImgixClient {
   constructor(opts = {}) {
-    this.settings = { ...DEFAULT_OPTIONS, ...opts };
+    this._settings = { ...DEFAULT_OPTIONS, ...opts };
     // a cache to store memoized srcset width-pairs
     this.targetWidthsCache = {};
-    if (typeof this.settings.domain != 'string') {
-      throw new Error('ImgixClient must be passed a valid string domain');
+
+    validateDomainString(this._settings.domain);
+
+    if (this._settings.includeLibraryParam) {
+      const defaultLibraryParam = 'js-' + ImgixClient.version();
+      this._settings.libraryParam =
+        this._settings.libraryParam || defaultLibraryParam;
     }
+  }
 
-    if (DOMAIN_REGEX.exec(this.settings.domain) == null) {
-      throw new Error(
-        'Domain must be passed in as fully-qualified ' +
-          'domain name and should not include a protocol or any path ' +
-          'element, i.e. "example.imgix.net".',
-      );
+  static version() {
+    return VERSION;
+  }
+
+  get settings() {
+    return this._settings;
+  }
+  get domain() {
+    return this.settings.domain;
+  }
+
+  set domain(value) {
+    validateDomainString(value);
+    this.settings.domain = value;
+  }
+
+  get includeLibraryParam() {
+    return this.settings.includeLibraryParam;
+  }
+
+  set includeLibraryParam(value) {
+    if (typeof value !== 'boolean') {
+      throw 'includeLibraryParam must be a boolean';
     }
+    this.settings.includeLibraryParam = value;
+  }
+
+  get libraryParam() {
+    return this.settings.libraryParam;
+  }
 
-    if (this.settings.includeLibraryParam) {
-      this.settings.libraryParam = 'js-' + ImgixClient.version();
+  set _libraryParam(value) {
+    if (typeof value !== 'string') {
+      throw 'libraryParam must be string';
     }
+    this.settings.libraryParam = value;
+  }
 
-    this.settings.urlPrefix = this.settings.useHTTPS ? 'https://' : 'http://';
+  get secureURLToken() {
+    return this.settings.secureURLToken;
   }
 
-  static version() {
-    return VERSION;
+  set secureURLToken(value) {
+    validateURLToken(value);
+    if (typeof value === 'undefined') {
+      this.settings.secureURLToken = undefined;
+    } else {
+      this.settings.secureURLToken = value;
+    }
```

Comment:
Just double checking, is this behavior consistent with what we were doing before?

Reply:
Good question, yes you were always able to set `secureURLToken` to undefined.

However, I think this conditional was left over from some debugging, my apols. When `value` is undefined, we don't need to confirm it is in fact `undefined` before setting the value of `secureURLToken`.